docgen/src/guides/prepare-for-v3.md

```diff
@@ -0,0 +1,42 @@
+---
+title: Prepare for v3
+mainTitle: Guides
+layout: main.pug
+category: guides
+withHeadings: true
+navWeight: 0
+editable: true
+githubSource: docgen/src/guides/prepare-for-v3.md
+---
+
+InstantSearch has always worked with the [Algolia search client](https://github.com/algolia/algoliasearch-client-javascript) behind the scenes. This will no longer be true for InstantSearch 3. To ease the migration from 2.8.x to 3.x.x, we recommend applying the changes from this guide now.
```

Comment:
This will be only a part of this guide, we should probably add a title here like: `Initializing InstantSearch`

Reply:
> To ease the migration from 2.8.x to 3.x.x, we recommend applying the changes from this guide now.

This could be used as a general introduction to the guide.

Also mention 2.x, instead of 2.8.x

Reply:
`version 2 to version 3` is another, simpler option

Reply:
If a user from `<= 2.8.0` follows this guide, it will break though (as `searchClient` won't be available before `2.8.0`). Should we ignore that?

Reply:
maybe a different sentence in the beginning saying that the guide will only work from 2.8 onwards

Reply:
Actually that's pretty good as an intro for the whole guide:

> Starting with 2.8, we are introducing changes and deprecations to prepare for the upcoming InstantSearch.js 3

WDYT?

Reply:
Do you mean instead of "InstantSearch has always worked [...] from this guide now." or before the current intro?

Reply:
> Starting with 2.8, we are introducing changes and deprecations to prepare for the upcoming InstantSearch.js 3. InstantSearch has always worked with the [Algolia search client](https://github.com/algolia/algoliasearch-client-javascript) behind the scenes. This will no longer be true for InstantSearch 3. To ease that migration, we recommend applying the changes from this guide now.

Reply:
Yep, that's what I've got right now. Wanted confirmation, thank you!